### DISCUSSION TITLE
Running sleap via import

### DISCUSSION THREAD

**Discussion by katiedunkley** [talmolab/sleap#1086] [Help!] [answered]
Hello, I am trying to follow this thread (https://github.com/talmolab/sleap/issues/436) as I am attempting to import centroid coordinates of individuals into sleap. However, when attempting to import sleap (e.g. on Jupyter/Google collab) I am failing. On Jupyter "import sleap" kills and resets the kernal as a fatal error, whilst with google collab I get the following error: "ImportError: cannot import name '_Union' from 'typing' (/usr/lib/python3.8/typing.py)". I'm working with a Mac with an M1 processor and the sleap GUI works (following the installation guidance). As a side note, I have also tried importing example data (e.g. h5 format, fly from https://zenodo.org/record/3366908#.Y6INIuzP1wQ) from deepposekit and that does not work either (error 'pose missing'), so I am little stuck. Thanks in advance.

**Comment by roomrys** [comment, marked as answer]
Hi @katiedunkley,

Based on [some research](https://github.com/aws/jsii/issues/913), one hypothesis is that there might be a conflict with the python version (`python>=3.8`) and one of our dependencies `cattrs`. However, I don't believe that is necessarily the case since I just tested a minimal Colab notebook:
```
!pip install sleap
exit()
```
```
import sleap
```
with python version 3.8.16 (both in Chrome and Microsoft Edge on Windows 10)
```
!python -V
```
resulting in no errors. Although the research example linked above used python v3.8.0....

Do you mind running the `!python -V` command in Colab (I would think Colab would default to the same python version across browsers/OS's, but just to check our basis). Also, if it isn't too much of a hassle, do you mind posting the full error messages and commands leading up to those errors?

Thanks,
Liezl

**Reply by katiedunkley** [reply]
Hi Liezl, 

Thank you very much for your reply - that seems to be working now - I can import sleap with no errors following the code above. I just tried running the script again (https://github.com/talmolab/sleap/issues/436) for a sanity check and it doesn't work still but that is using the !pip install sleap==1.1.0a7 which could be where it's going wrong for me. Many thanks for the help. 

Best wishes, 
Katie
